### PR TITLE
Add scroll-spy functionality

### PR DESCRIPTION
**Pull request by rogeruiz**
@maya @juliaelman @yozlet

### CODE REVIEW

**Comment by maya** [comment]
Isn't working for research report for me in Chrome when I click on it: 



Update: sometimes it works sometimes it doesn't?

**Comment by maya** [comment]
@rogeruiz jk just tried again and that issue seems to be fine!

**Comment by maya** [comment]
I know @carodew says she tested the Galaxy S5, but these were my results:



Here's the iPhone 6 (everything except overview works on click):



Large screens seem :+1: for me.

**Comment by rogeruiz** [comment]
Hey @maya Thanks for the screens. I think I cleared up the bugs related to the highlighting. I'll point them out in the line-notes.

File: assets/scripts/components/scroll-spy.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+var $nav = $('.js-sticky-nav');
+var originalNavigationHeight = $nav.outerHeight(true);
+
+function calculateSections ( $links ) {
+  // Get all links & sections and serialize them
+  //
+  return [].map.call( $links, function (el) {
+    var $link = $(el);
+    var $section = $($link.attr('href'));
+    var topOffset = $section.position().top - originalNavigationHeight;
```
Comment (rogeruiz) [inline assets/scripts/components/scroll-spy.js:13]: Make sure that the sticky navigation's height is taken into account for the `topOffset` of all sections.

File: assets/scripts/components/scroll-spy.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+module.exports = function scrollSpy () {
+
+  var $links = $('.js-scroll-spy').find('a[href^="#"]');
+  var sections = calculateSections($links);
+
+  var scrollY = $(window).scrollTop();
+
+  var remainingHeight = $(document).height() - scrollY - window.innerHeight;
```
Comment (rogeruiz) [inline assets/scripts/components/scroll-spy.js:31]: Calculate the remaining height of the page based on the current scroll y-position.

File: assets/scripts/components/scroll-spy.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  $links.removeClass('is-active');
+
+  sections.forEach(function (section, index) {
+
+    if (section.topOffset <= scrollY &&
+        section.bottomOffset >= scrollY) {
+      section.$link.addClass('is-active');
+    }
+
+    if (remainingHeight < window.innerHeight * 0.25) {
```
Comment (rogeruiz) [inline assets/scripts/components/scroll-spy.js]: If the remaining height of the page is less than 25% of the height of the page, the user will never be able to scroll to meet the `topOffset` calculated above. So, remove all the `is-active` classes and apply `is-active` to the last item in the `sections` array.

File: assets/styles/homepage.scss [hunk 4]
```diff
@@ -268,6 +272,7 @@ h2 {
 // Contact section extra padding so scrollspy works
 .section-end {
   @include media ($medium-screen) {
+    padding-top: 13rem;
```
Comment (rogeruiz) [inline assets/styles/homepage.scss]: I included some extra padding on the top of the Contact section to visually balance the contact section. It was too top heavy before. I'm okay with reverting this change, NBD.

File: assets/scripts/components/scroll-spy.js (hunk 3 above)
Comment (rogeruiz) [inline assets/scripts/components/scroll-spy.js]: This assumption works for this specific navigation and sections number and isn't suited to other kinds of navigation / section combinations.

File: assets/styles/homepage.scss (hunk 4 above)
Comment (maya) [inline assets/styles/homepage.scss]: I would not add this, to keep it consistent with the others. Let's change it back.

Comment (rogeruiz) [inline assets/styles/homepage.scss]: Sounds good @maya I'll remove it.

**Comment by maya** [comment]
iPhone 6 and Galaxy S5 look good to me!

**Comment by rogeruiz** [comment]
Hey @juliaelman or @msecret. Could one of you take a look at the JS for this PR? @maya and I have tested it in all major browsers. Thanks!

**Comment by msecret** [comment]
LGTM

**Comment by maya** [comment]
:rocket:

**Pull request merged**